pkg/gitreceive/git/sha.go

```diff
@@ -0,0 +1,31 @@
+package git
+
+import (
+	"fmt"
+)
+
+const (
+	// this constant represents the length of a shortened git sha - 8 characters long
+	shortShaIdx = 8
+	fullShaLen  = 40
+)
+
+type ErrGitShaTooShort struct {
+	sha string
+}
+
+func (e ErrGitShaTooShort) Error() string {
+	return fmt.Sprintf("git sha %s was too short", e.sha)
+}
+
+type SHA struct {
+	Full  string
+	Short string
+}
+
+func NewSha(rawSha string) (*SHA, error) {
+	if len(rawSha) < fullShaLen {
```

Comment:
Fixed in https://github.com/arschles/builder/commit/0ff588721161a854c4496d733602b493fdf3f49f